### DISCUSSION TITLE
Warning option when moving currently used directories (suggestion/bug)

### DISCUSSION THREAD

**Discussion by Podloot** [godotengine/godot-proposals#6498] [Editor]
Hello,

It is way too easy to move a directory to another location in the FileSysten panel, increasing the possibility of a accidental mouse slip (happened to me more often than I would like to admit). The issue with this is, that if have any scene open in a directory that is moved by accident, Godot crashes and you can't undo the action (at least for me). After opening the project again, and (intuitivly, your first thought, but not the best idea) moving the directory back to its place, some scenes are likely to be corrupted (export variables and scripts (c#) seem to have a wrong location).

I have a couple suggestion to solve this issue (even if the chrasing bug is fixed):
- You have to hold the mousebutton down for 3/5 seconds to move it.
- The action gets canceled / you get a warning when you want to move a directory with an opened scene in it
          - The scenes are saved and closed when you move a folder (?)
- You get a warning when you want to move a directory with any scene in it.
- Remove this feature and make it only possible with the right-click -> move to option (or set is an option in the settings menu)

I hope this gets implemented somehow, it would save me a lot of trouble. :) (or moving directories should be made more robust, and less likely to corupt files when done incorreclty).

Godot: Godot v4.0 - Stable Mono version Win64, Windows 11

Kind regards, Podloot.

**Comment by Calinou** [comment]
It makes sense to add a confirmation dialog if moving a non-empty folder. I'm not sure if checking the number of users for each resource within the folder (*recursively*) can be done quickly.

**Reply by Zireael07** [reply]
Yeah, I have the same issue as OP. Please please.
